docs/content/users/basics/how-ddev-works.md

```diff
@@ -1,21 +1,21 @@
-* The optional `dba` container runs PhpMyAdmin for projects with MySQL or MariaDB.
+* The `ddev-webserver` container (one per project) runs `nginx` or `apache` and `php-fpm` for a single site, so it does all the basic work of a PHP-interpreting web server.
+* The `ddev-dbserver` container (one per project) handles MariaDB/MySQL/PostgreSQL database management. It can be reached from the web server by the hostname `db` or with the more explicit name `ddev-<projectname>-db`.
+* The optional `dba` container runs phpMyAdmin for projects with MySQL or MariaDB.
 * Additional add-on services may be there for a given project, for example `solr` or `elasticsearch` or `memcached`.
 
-Although it's not common usage, different projects can communicate with each other as described in the [FAQ](faq.md#projects-communicate-with-each-other)
+Although it’s not common usage, different projects can communicate with each other as described in the [FAQ](faq.md#projects-communicate-with-each-other).
 
-Now for the two oddball containers, which are global (there is only one of each).
+Now for the two oddball global containers (there’s only one of each):
 
-* The `ddev-router` container is a "reverse proxy". It takes incoming HTTP/S requests and looks up the hostname in the incoming URL and routes it to the correct project's `ddev-webserver`. Depending on the project's configuration with `additional_hostnames` and `additional_fqdns` it can route many different URLs to a single project's `ddev-webserver`. If like most people you use the named URLs (like `https://something.ddev.site`) then your request goes through the router. When you use the `127.0.0.1` URLs, the requests go directly to the `ddev-webserver`.
-* The `ddev-ssh-agent` container runs an `ssh-agent` inside the docker network so that after you do a `ddev auth ssh` all the different projects can use your ssh keys for outgoing requests (like private composer access or scp from a remote host).
+* The `ddev-router` container is a “reverse proxy”. It takes incoming HTTP/S requests, looks up the hostname in the incoming URL, and routes it to the correct project’s `ddev-webserver`. Depending on the project’s configuration with `additional_hostnames` and `additional_fqdns`, it can route many different URLs to a single project’s `ddev-webserver`. If, like most people, you use the named URLs like `https://something.ddev.site`, your request goes through the router. When you use the `127.0.0.1` URLs, the requests go directly to the `ddev-webserver`.
+* The `ddev-ssh-agent` container runs an `ssh-agent` inside the Docker network so that after you do a `ddev auth ssh` all the different projects can use your SSH keys for outgoing requests—like private Composer access or SCP from a remote host.
 
-Here's a basic diagram of how it works inside the docker network:
+Here’s a basic diagram of how it works inside the Docker network:
 
 ![DDEV Docker Network Architecture](../topics/images/DDEV%20Container%20Architecture.svg)
```

Comment:
It would be great to turn this into a [mermaid](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-diagrams) diagram of course. Future!

Reply:
> Through troubleshooting. Thanks! Wow, is that a bloated thing!

I’ve found one of the strengths of DDEV’s docs (and GitHub issues) to be that I can usually find some helpful, existing description or discussion of the issue I’m having. I love that the Troubleshooting section is actually thorough, and would only try to make it easier to navigate. I’d also remove redundant tips from elsewhere in the docs in favor of directly linking to the Troubleshooting page, which (hopefully) puts emphasis in the right place and eases maintenance.

Reply:
I think some of it can be tabbed up and that could help some. And some kind of flow chart might improve it. 

A lot of things in the docs are there just so that people who know the answer can just provide an authoritative link instead of repeating the same thing over and over again.

Reply:
I’m a little wary of tab use going too far and making it easier to get lost, but it seems like we’re thinking along the same lines. 👍

Reply:
Just thinking out loud. Obviously that page has too much content for an ordinary person to survive reading. However, there's some skimmability, and certainly the ability for a knowledgable person to provide a link. Oh, and the search is working *awesome* these days. So amazing.

Reply:
> Obviously that page has too much content for an ordinary person to survive reading.

😆 

> Oh, and the search is working *awesome* these days. So amazing.

I’ve appreciated it as a casual reader—well done! I know firsthand it can be tricky.